Code.Rmd

```diff
@@ -131,37 +259,148 @@ There are some functions that modify global settings that you should never use b
   If you're using `source()` to create a dataset, instead switch to `data/`
   as described in [datasets](#data).
 
-Other functions need to be used with caution.
-If you use them, make sure to clean up after yourself with `on.exit()`:
+#### Managing state
 
-* If you modify global `options()` or graphics `par()`, save the old values 
-  and reset when you're done:
-  
-    ```{r, eval = FALSE}
-    old <- options(stringsAsFactors = FALSE)
-    on.exit(options(old), add = TRUE)
-    ```
+Sometimes you cannot avoid modifying the state of the world, in which case you just have to make sure that you put things back the way you found them.
 
-* Avoid modifying the working directory. If you do have to change it, make sure
-  to change it back when you're done:
+The first function to know about is base R's [`on.exit()`](https://rdrr.io/r/base/on.exit.html).
```

Comment:
I think it would be better to start with the specific withr functions, then talk about `withr::defer()`, and then link to something that explains `on.exit()`. The only downside is that might force a discussion about dependencies earlier than you might otherwise want it.